### PR TITLE
improve: Explicitly indicate gzip support in provider config

### PR DESCRIPTION
**Pull request by pxrl**
This seems to be a client-side flag to the RPC that it will accept
gzip-compressed responses. Per the ethers documentation, this
flag actually defaults to false. The decision as to whether to
compress (or not) should be up to the server. Infura don't appear
to support it yet, but Alchemy are pretty positive about it:

> To provide users with better product experiences, we updated our
> internal infrastructure to offer Alchemy developers support for
> gzip compression on all responses larger than 1kb in size.

> Gzip compression offers up to a 95% decrease in the size of files
> sent over a streaming connection. However, the actual latency and
> bandwidth savings is dependent on the structure of data being
> broadcast, the connection speeds of the client and server, and the
> size of the response.

> In practice, we’ve seen roughly a 75% improvement in the total
> latency of typical JSON-RPC replayTransaction calls.

Note: I've tested that it works with a peasant-tier Alchemy plan, but
haven't actually checked to see whether it performs better yet. I'm
just placing this here in case anyone else wants to give it a try with
heavier loads.

Ref: ACX-67

### CODE REVIEW

**Comment by pxrl** [comment]
I am now reviving this request after I found that Alchemy provide [instructions for verifying the performance improvement with Gzip](https://docs.alchemy.com/docs/how-to-enable-compression-to-speed-up-json-rpc-blockchain-requests#test-the-json-rpc-response-latency-decrease-using-gzip-compression).

TLDR: Running with compression seems to consistently deliver performance that is on par or faster than without compression. As response sizes grow, the effect of compression improves (as is to be expected). Improvements have been seen against both Infura and Alchemy.

Test description:
I sourced some previous eth_getLogs requests that included actual data in the response payload, and instrumented curl (per Alchemy's instructions) to spit out some diagnostic information.

There is potential for bias in these results, depending on where the compression is applied. There is a chance that the first request might incur a lookup penalty, and that the 2nd request might bypass that penalty by being served a cached response. Despite this, some quick testing seems to show that running with compression is still consistently on par or better.

The following tests were conducted against Alchemy:

```
Testing request: {"method":"eth_getLogs","params":[{"fromBlock":"0xf1f146","toBlock":"0xf21e34","address":"0x4d9079bb4165aeb4084c526a32695dcfd2f77381","topics":["0x56450a30040c51955338a4a9fbafcf94f7ca4b75f4cd83c2f5e29ef77fbe0a3a"]}],"id":280,"jsonrpc":"2.0"}
First request, without Gzip
time_namelookup:  0.001185s
time_connect:  0.012794s
time_appconnect:  0.081071s
time_pretransfer:  0.081276s
time_redirect:  0.000000s
time_starttransfer:  0.081279s
------------------------------------------------------
size_download:  3340290B
time_total:  1.203167s


2nd request, with Gzip
time_namelookup:  0.000477s
time_connect:  0.012239s
time_appconnect:  0.067162s
time_pretransfer:  0.067762s
time_redirect:  0.000000s
time_starttransfer:  0.067770s
------------------------------------------------------
size_download:  344925B
time_total:  0.602024s
```
```
Testing request: {"method":"eth_getLogs","params":[{"fromBlock":"0xf1f146","toBlock":"0xf21e34","address":"0x4d9079bb4165aeb4084c526a32695dcfd2f77381","topics":["0x4a4fc49abd237bfd7f4ac82d6c7a284c69daaea5154430cff04ad7482c6c4254"]}],"id":279,"jsonrpc":"2.0"}
First request, without Gzip
time_namelookup:  0.001147s
time_connect:  0.011517s
time_appconnect:  0.089493s
time_pretransfer:  0.089705s
time_redirect:  0.000000s
time_starttransfer:  0.089708s
------------------------------------------------------
size_download:  336741B
time_total:  0.405226s


2nd request, with Gzip
time_namelookup:  0.001117s
time_connect:  0.012425s
time_appconnect:  0.069269s
time_pretransfer:  0.069481s
time_redirect:  0.000000s
time_starttransfer:  0.069485s
------------------------------------------------------
size_download:  53186B
time_total:  0.299228s
```

```
Testing request: {"method":"eth_getLogs","params":[{"fromBlock":"0xea0ff8","toBlock":"0xea5e18","address":"0x4d9079bb4165aeb4084c526a32695dcfd2f77381","topics":["0x0a21fdd43d0ad0c62689ee7230a47309a050755bcc52eba00310add65297692a"]}],"id":260,"jsonrpc":"2.0"}
First request, without Gzip
time_namelookup:  0.000561s
time_connect:  0.011916s
time_appconnect:  0.073674s
time_pretransfer:  0.074322s
time_redirect:  0.000000s
time_starttransfer:  0.074331s
------------------------------------------------------
size_download:  1849B
time_total:  0.233042s


2nd request, with Gzip
time_namelookup:  0.000691s
time_connect:  0.011823s
time_appconnect:  0.071673s
time_pretransfer:  0.072282s
time_redirect:  0.000000s
time_starttransfer:  0.072290s
------------------------------------------------------
size_download:  383B
time_total:  0.232423s
```


This test was conducted against Infura:

```
Testing request: {"method":"eth_getLogs","params":[{"fromBlock":"0xf1f146","toBlock":"0xf21e34","address":"0x4d9079bb4165aeb4084c526a32695dcfd2f77381","topics":["0x56450a30040c51955338a4a9fbafcf94f7ca4b75f4cd83c2f5e29ef77fbe0a3a"]}],"id":280,"jsonrpc":"2.0"}


1st request, with Gzip
time_namelookup:  0.000674s
time_connect:  0.102118s
time_appconnect:  0.367075s
time_pretransfer:  0.367847s`
time_redirect:  0.000000s
time_starttransfer:  0.368090s
------------------------------------------------------
size_download:  318378B
time_total:  1.186462s


2nd request, without Gzip
time_namelookup:  0.000987s
time_connect:  0.103095s
time_appconnect:  0.366027s
time_pretransfer:  0.366708s
time_redirect:  0.000000s
time_starttransfer:  0.366939s
------------------------------------------------------
size_download:  3340290B
time_total:  2.297369s
```

**Review by nicholaspai — APPROVED** [review]
